### PR TITLE
make select dumb

### PR DESCRIPTION
**Pull request by dmfenton**
This PR removes all logic from the select function except getting feature collections from the database. 

This is a breaking change.

### CODE REVIEW

**Comment by dmfenton** [comment]
cc @koopjs/dev

**Comment by dmfenton** [comment]
@chelm I couldn't find the use of idFilter in any of the existing Koop providers. Can you show me where/how its used?

**Comment by chelm** [comment]
@dmfenton sure. It is super crucial and completely my fault for not docing it i guess. However, its also super concerning that it was removed, this would have been an issue pretty quickly I think. One sec.

**Comment by chelm** [comment]
@dmfenton its used in the Exporter for dealing with large data. 

https://github.com/koopjs/koop/blob/master/lib/Exporter.js#L29-L44

This was introduced as an optimization in place of using `offset` & `limit` in selecting pages of data from the db. It could very well be moved into the Cache itself (which is probably where it belongs honestly as koop-pgcache is probably the only that would use it). your call...

**Comment by chelm** [comment]
also remember that typically providers will never call pgcache or any cache directly, they will always route through code in koop/lib in some way. So any breaking changes in the caches should be protected from the providers and handled more centrally.

**Comment by dmfenton** [comment]
> @dmfenton its used in the Exporter for dealing with large data.

Good looking out. Really glad you were watching to catch that one. I'll add it back. We can consider a more general solution in the future.

**Comment by chelm** [comment]
+1 for changelogs...

**Comment by dmfenton** [comment]
And we're back

**Comment by dmfenton** [comment]
@chelm @ngoldman any more changes to make?

File: index.js [hunk 1]
```diff
@@ -295,6 +281,7 @@ module.exports = {
    * @returns {string} sql
    */
   createWhereFromSql: function (where, fields) {
+    if (where === '1=1') return where
```
Comment (ungoldman) [inline index.js:284]: smh esri

**Comment by ungoldman** [comment]
doc looks good and tests are passing... lgtm. unless @chelm has ideas I say we push the green button.

**Comment by chelm** [comment]
lgtm :baby:

**Pull request merged**